### PR TITLE
Remove ServiceMeshPolicy and SMRbacConfig for Maistra installations

### PR DESCRIPTION
**Pull request by xeviknal**
Maistra is no longer using those objects. This PR is meant to remove those references from the frontend.

requires https://github.com/kiali/kiali/pull/3256

### CODE REVIEW

**Comment by jmazzitelli** [comment]
Do we need to remove this: https://github.com/kiali/helm-charts/blob/master/kiali-server/templates/role.yaml#L64-L83
and all the other places where those are (e.g. the OLM metadata golden copies?)

**Review by hhovsepy — APPROVED** [review]

File: src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx [hunk 1]
```diff
@@ -430,7 +430,7 @@ class IstioConfigDetailsPage extends React.Component<RouteComponentProps<IstioCo
     const istioStatusMsgs = this.getStatusMessages();
     const objectReferences = this.objectReferences();
     const refPresent = objectReferences.length > 0;
-    const showCards = refPresent || this.hasOverview() || istioStatusMsgs.length > 0;
+    const showCards = refPresent || this.hasOverview() || (!!istioStatusMsgs && istioStatusMsgs.length > 0);
```
Comment (lucasponce) [inline src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx:433]: Not sure if I understand this validation, `istioStatusMsgs` will be always an array, or perhaps I'm missing some case.

**Review by lucasponce — APPROVED** [review]
LGTM
But it requires a couple of minor changes in the backend one.

File: src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx (hunk 1 above)
Comment (xeviknal) [reply src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx:433]: It is pretty unexpected. Testing around, I came across a nasty bug. It was a length access to a undefined variable.

**Pull request merged**